src/app/shell/admin-tools/data/admins/create-admin/create-admin.component.ts

```diff
@@ -120,17 +161,22 @@ export class CreateAdminComponent extends CreateFormComponent implements OnInit,
     const dialogRef = this.matDialog.open(ConfirmationModalWindowComponent, {
       width: Constants.MODAL_SMALL,
       data: {
-        type: (this.adminRole === AdminRole.ministryAdmin) ? ModalConfirmationType.createMinistryAdmin : undefined
+        type: this.editMode? ModalConfirmationType.updateMinistryAdmin : ModalConfirmationType.createMinistryAdmin
       }
     });
 
     dialogRef.afterClosed().subscribe((result: boolean) => {
       if (result) {
-        let ministryAdmin = new MinistryAdmin(
+        const ministryAdmin = new MinistryAdmin(
           this.AdminFormGroup.value, 
-          this.AdminFormGroup.get('institution').value.id
-          );
-        this.store.dispatch(new CreateMinistryAdmin(ministryAdmin));
+          this.AdminFormGroup.get('institution').value.id,
+          this.adminId
+        );
+        if (this.editMode) {
+          this.store.dispatch(new UpdateMinistryAdmin(ministryAdmin));
```

Comment:
this.store.dispatch(this.editMode? new UpdateMinistryAdmin(ministryAdmin) : new CreateMinistryAdmin(ministryAdmin));